### ISSUE TITLE
comment start sequences

### ISSUE THREAD

**Issue by DanGrayson** [Macaulay2/M2#674]
The reason our comment coloring in emacs doesn't work well is that it's a hodge-podge of
manual coloring and syntax-table use.  Switching to using just the syntax table would
be better, probably, but then we run afoul of the emacs capability: it offers two 2-character
comment sequences, but their first characters must be the same, see below.  So the problem
is that "{*" and "--" don't start with the same character.

What should we do?  Replace "{*" and "*}" with something else?

```
   Here is a table of the possible flags for a character C, and what
they mean:

   * `1' means C is the start of a two-character comment-start sequence.

   * `2' means C is the second character of such a sequence.

   * `3' means C is the start of a two-character comment-end sequence.

   * `4' means C is the second character of such a sequence.

   * `b' means that C as a comment delimiter belongs to the alternative
     "b" comment style.

     Emacs supports two comment styles simultaneously in any one syntax
     table.  This is for the sake of C++.  Each style of comment syntax
     has its own comment-start sequence and its own comment-end
     sequence.  Each comment must stick to one style or the other;
     thus, if it starts with the comment-start sequence of style "b,"
     it must also end with the comment-end sequence of style "b."

     The two comment-start sequences must begin with the same
     character; only the second character may differ.  Mark the second
     character of the "b"-style comment-start sequence with the `b'
     flag.

     A comment-end sequence (one or two characters) applies to the "b"
     style if its first character has the `b' flag set; otherwise, it
     applies to the "a" style.

     The appropriate comment syntax settings for C++ are as follows:

    `/'
          `124b'

    `*'
          `23'

    newline
          `>b'

     This defines four comment-delimiting sequences:

    `/*'
          This is a comment-start sequence for "a" style because the
          second character, `*', does not have the `b' flag.

    `//'
          This is a comment-start sequence for "b" style because the
          second character, `/', does have the `b' flag.

    `*/'
          This is a comment-end sequence for "a" style because the first
          character, `*', does not have the `b' flag.

    newline
          This is a comment-end sequence for "b" style, because the
          newline character has the `b' flag.
```

**Comment by DanGrayson** [comment]
Proposal : change `{* ... *}` to `-* ... *-`.

**Comment by DanGrayson** [comment]
resolved by commit f2c87ac67898b082eee96ec4c9df18b10a400e15

**Comment by DanGrayson** [comment]
PS: Our block comments do not nest, because then we'd have to parse the contents completely.

**Comment by DanGrayson** [comment]
I'm reopening this issue so we can figure out what to do about colorizing comment sequences delimited by /// ... /// .  It can be annoying when it doesn't work out!

**Comment by DanGrayson** [comment]
Here is the documentation for /// ... ///:

```
i3 : help "///"

o3 = /// -- delineate a string with slashes
     **************************************

     This method for entering a string involves no escape characters, so it can be used
     for easily inserting large chunks of text into a string without treating the
     characters \ and " specially.  A series of more than 3 slashes can be represented
     within the string by doubling all but the last two, and a series of 1 or more
     slashes can be represented at the end of the string by doubling each of them; this
     allows an arbitrary string to be represented.
```

All we have to do is to come up with a regular expression to match those comments, and put it into 'emacs/makesyms.m2'.

**Comment by DanGrayson** [comment]
... and here it is:
```
"///\\(/?/?[^/]\\|\\(//\\)*////[^/]\\)*\\(//\\)*///"
```

It says that such a comment starts with ///, then it consists of a sequence of any of
these substrings
-   an optional /, and optional /, and something not a /
-   a sequence of copies of // followed by ////
Then it ends with a sequences of copies of // followed by ///.

**Comment by DanGrayson** [comment]
Resolved by commit dfad7d64bdcfd29aa342345f80bef98bd9b6ab0b and commit cd648bae5454967cefb21880091429c6c6e712b1
